src/lib/tls/tls_policy.cpp

```diff
@@ -192,13 +192,22 @@ std::vector<Group_Params> Policy::key_exchange_groups() const {
 }
 
 std::vector<Group_Params> Policy::key_exchange_groups_to_offer() const {
-   // by default, we offer a key share for the most-preferred group, only
-   std::vector<Group_Params> groups_to_offer;
-   const auto supported_groups = key_exchange_groups();
-   if(!supported_groups.empty()) {
-      groups_to_offer.push_back(supported_groups.front());
+   /*
+   By default, we offer a key share for the most-preferred pure ECC group.
+
+   We skip PQC (or hybrids) since the keys are much larger and they are not
+   yet widely supported; the most common case is we waste a lot of packet
+   space sending a key share that the peer will ignore.
+
+   Likewise we skip DH since the keys are large
+   */
+   for(auto group: key_exchange_groups()) {
+      if(group.is_pure_ecc_group()) {
+         return {group};
+      }
    }
-   return groups_to_offer;
+   // Send no initial key share
+   return {};
```

Comment:
Mhh, frankly, if we reach here, we should probably just offer whatever is the "most preferred group" instead of not offering anything.

I'm thinking of specialized applications that configure a short list of non-ECC groups for some reason. They would always end up with a HelloRetryRequest roundtrip for no obvious reason. Sure, such applications could override this policy... but do they care? And how long will it take them to notice the performance regression otherwise?